mcs/class/System/System/Uri.cs

```diff
@@ -170,9 +170,16 @@ protected Uri (SerializationInfo serializationInfo, StreamingContext streamingCo
 			}
 		}
 
+		// When used instead of UriKind.RelativeOrAbsolute paths such as "/foo" are assumed relative.
+		const UriKind DotNetRelativeOrAbsolute = (UriKind) 4;
+
 		public Uri (string uriString, UriKind uriKind)
 		{
 			source = uriString;
+
+			if (uriKind == DotNetRelativeOrAbsolute)
+				uriKind = (uriString.StartsWith ("/"))? UriKind.Relative : UriKind.RelativeOrAbsolute;
```

Comment:
uriString can be null
use StringComparison.Ordinal if you need to use StartsWith